### PR TITLE
E2E tests: dump failure information

### PR DESCRIPTION
**Pull request by fedepaol**
Dump relevant information when a test fails.

### CODE REVIEW

File: e2etests/pkg/k8s/pods.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+const labelSelector = "app=frr-k8s"
+
+// SpeakerPods returns the set of pods running the speakers.
```
Comment (liornoy) [inline e2etests/pkg/k8s/pods.go:16]: nit: need to update theis comment

File: e2etests/pkg/k8s/pods.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	}
+	if len(pods.Items) == 0 {
+		return nil, errors.New("No frrk8s pods found")
+	}
+	speakerPods := make([]*corev1.Pod, 0)
+	for _, item := range pods.Items {
+		i := item
+		speakerPods = append(speakerPods, &i)
+	}
+	return speakerPods, nil
```
Comment (liornoy) [inline e2etests/pkg/k8s/pods.go:32]: and the `speakerPods` var name

File: e2etests/pkg/dump/k8s.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	reporter, err := k8sreporter.New(kubeconfig, addToScheme, dumpNamespace, ReportPath, crds...)
+	if err != nil {
+		log.Fatalf("Failed to initialize the reporter %s", err)
+	}
+	return reporter
+}
+
+func K8sInfo(testName string, reporter *k8sreporter.KubernetesReporter) {
+	reporter.Dump(10*time.Minute, testName)
```
Comment (liornoy) [inline e2etests/pkg/dump/k8s.go:53]: can you please pass the 10 minutes as a const for clarity?
we named this as `LogsExtractDuration` on the cnf-tests.

File: e2etests/pkg/dump/frr.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+	"go.universe.tf/metallb/e2etest/pkg/executor"
+	"go.universe.tf/metallb/e2etest/pkg/frr"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+func BGPInfo(testName string, FRRContainers []*frrcontainer.FRR, cs clientset.Interface, f *framework.Framework) {
+	if ReportPath == "" {
+		log.Fatalf("ReportPath is not set")
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply e2etests/pkg/dump/frr.go:23]: Fatalf comes with an embedded panic. I will revert to use the framework logging and will panic in the line below

**Comment by liornoy** [comment]
lgtm. I validated the reports locally on my end.

**Pull request merged**